### DISCUSSION TITLE
GraphCache updates.

### DISCUSSION THREAD

**Discussion by yaldram** [urql-graphql/urql#3226] [Q&A]
Hello Community, I hope you are doing well, I have a small query, I have a small todo CRUD application, when I add a new Todo the mutation executes successfully, but I cannot see the new added Todo, therefore I used the `@urql/exchange-graphcache` library and updated the cache locally saving a network request. 

But now the `delete` mutation does not update the UI automatically when I delete the todo, is this expected ? So if I use `@urql/exchange-graphcache` do I have to handle delete case for all queries, thanks.

**Comment by sviat-barbutsa** [comment]
Basically, yeah, you need to create an updater for the mutation like described here:
[Cache Update](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/)
 or you can just simply re-fetch the query and get the actual data from the server hovewer it depends on your app scenario
